### ISSUE TITLE
Very severe and strange map-gen corruption

### ISSUE THREAD

**Issue by axlegear** [Blood-Asp/GT5-Unofficial#290]
Since the latest update, there have been some strange chunk mis-gens, many corrupt and causing crashes and server lag or stalls.  Today one generated in a tundra that had trees made of redstone engines and veins of random crap, and invalid blocks that if looked at in WAILA would crash everything and prevent the server from reloading.  Deleting the chunk does no good, it just generates a new chunk that is also corrupt.

I am trying to attach the logs but github won't let me without write permission...

**Comment by axlegear** [comment]
http://sharesend.com/zb5cilpd - PlayerActivity.txt
http://sharesend.com/i2z1w0vj - fml-junk-earlystartup.txt
http://sharesend.com/q9oe8ht1 - fml-server-1.txt
http://sharesend.com/3xlyboyl - fml-server-2.txt
http://sharesend.com/sfwdjgkr - fml-server-3.txt
http://sharesend.com/lrs9skzr - fml-server-latest.txt
http://sharesend.com/vbm6q8mb - GregTech.txt
http://sharesend.com/4tehy7uj - latest.txt

**Comment by axlegear** [comment]
https://drive.google.com/file/d/0B7PNQy2eWIcYSXRpRkxyN1ZCSU0/view?usp=sharing

**Comment by Blood-Asp** [comment]
I saw something like that before and know it happens really, really rarely. I have no idea were that comes from and it corrupts usually only 1-2 chunks.

There is nothing that shows it comes from gregtech. Might come from any other mod in your pack. So i can't accept this issue as long there is nothing that somehow points to anything i can do.

**Comment by axlegear** [comment]
The last thing the server said before it crashed was that it was skipping a GT block thousands of times.

Also, this corruption happens only when GT5 is present.  When I removed it and deleted GT content in-world, the chunk generated correctly

Lastly, 'rare' or not, it happens consistently now in new chunks are generated, regardless of whether I delete it or change the seed.

If you can't identify the problem at GT, at least try to help me find some insight, because it has effectively ruined the server.

**Comment by Blood-Asp** [comment]
I find two execeptions in the latest log: stacksonstacks null pointer exception on loading.
buildcraft engines stack overflow exection (likely because too many were loaded at once.)
The "skipping gt blocks" happens just because of the chunkerror shifting ids to some GT blocks were they get removed as invalid. These do not come from GT itself.

**Comment by Blood-Asp** [comment]
Also, if you say reverting to old GT version stops it, do you have to go back to GT 5.09.08, or is 5.09.09 still fine?

**Comment by Blood-Asp** [comment]
Oh, i also see fastcraft in the stacktrace...

**Comment by Blood-Asp** [comment]
So instead of only downgrading GT, try to also remove other mods and see if it stops. Maybe it is some strange mod interaction. Otherwise, the allready 2 weeks out GT version would have resulted into quite a lot more bugreport if it were to happen with GT only.

**Comment by axlegear** [comment]
We weren't on the 5.09.09 version long, so I went to 5.09.08.  I'll start testing mod by mod, but I can't figure what mod it could be.  The only mods other than GT that changed were Buildcraft and Pam's Harvestcraft.  This will be a very long, slow, and painful process of testing... so I had hoped some insight on your part may help.

**Comment by Blood-Asp** [comment]
You can speed up the process quite a lot. Allways use one half of the mods. 
First remove one half and see if it is still there. If yes, readd them and remove other half. If it is gone then (and GT is in both times), then again readd half of the removed mods. Using that way, at best 7 iteration = 14 restarts are needed for checking all mods.

**Comment by Blood-Asp** [comment]
And as i said, i would start testing with the removal of the 3 involved mods: Fastcraft, Buildcraft and Stacksonstacks.

**Comment by axlegear** [comment]
I'll start with Buildcraft since it's known to be problematic more and more over the past few months.  SoS is new and it doesn't generate any content at all in the world, so I doubt that.  Ditto for fastcraft, but I am unsure how it modifies stuff in general, so it's second on the list.

**Comment by axlegear** [comment]
Removal of these mods stopped the redstone engine trees, but there are still invalid blocks which, if looked at, crashes the server.  It looks to be a beehive.
Removing forestry next.

**Comment by axlegear** [comment]
Forestry was removed but it's still there.  It looks like a beehive.  I'm going to do further testing.

**Comment by axlegear** [comment]
Removing every mod except gregtech now.

**Comment by axlegear** [comment]
Still there.  I'm gonna see if I can't dig a way to identify the bad block.

**Comment by axlegear** [comment]
Not sure how else to try and ID the block.. turning off Waila and picking it up via dupe results in a crash.

---- Minecraft Crash Report ----
// Why did you do that?

Time: 10/31/15 4:38 PM
Description: Rendering item

java.lang.NullPointerException: Rendering item
	at net.minecraft.client.renderer.RenderBlocks.func_147768_a(RenderBlocks.java:7021)
	at ic2.core.block.personal.RenderBlockPersonal.renderInventoryBlock(RenderBlockPersonal.java:31)
	at cpw.mods.fml.client.registry.RenderingRegistry.renderInventoryBlock(RenderingRegistry.java:125)
	at FMLRenderAccessLibrary.renderInventoryBlock(FMLRenderAccessLibrary.java:59)
	at net.minecraft.client.renderer.RenderBlocks.func_147800_a(RenderBlocks.java:8145)
	at net.minecraft.client.renderer.entity.RenderItem.renderItemIntoGUI(RenderItem.java:446)
	at net.minecraft.client.renderer.entity.RenderItem.func_82406_b(RenderItem.java:565)
	at net.minecraft.client.gui.GuiIngame.func_73832_a(GuiIngame.java:955)
	at net.minecraftforge.client.GuiIngameForge.renderHotbar(GuiIngameForge.java:206)
	at net.minecraftforge.client.GuiIngameForge.func_73830_a(GuiIngameForge.java:141)
	at net.minecraft.client.renderer.EntityRenderer.func_78480_b(EntityRenderer.java:1038)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1001)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at net.minecraft.client.renderer.RenderBlocks.func_147768_a(RenderBlocks.java:7021)
	at ic2.core.block.personal.RenderBlockPersonal.renderInventoryBlock(RenderBlockPersonal.java:31)
	at cpw.mods.fml.client.registry.RenderingRegistry.renderInventoryBlock(RenderingRegistry.java:125)
	at FMLRenderAccessLibrary.renderInventoryBlock(FMLRenderAccessLibrary.java:59)
	at net.minecraft.client.renderer.RenderBlocks.func_147800_a(RenderBlocks.java:8145)
	at net.minecraft.client.renderer.entity.RenderItem.renderItemIntoGUI(RenderItem.java:446)

-- Item being rendered --
Details:
	Item Type: ic2.core.item.block.ItemPersonalBlock@1b5f9430
	Item Aux: 6
	Item NBT: null
	Item Foil: false
Stacktrace:
	at net.minecraft.client.renderer.entity.RenderItem.func_82406_b(RenderItem.java:565)
	at net.minecraft.client.gui.GuiIngame.func_73832_a(GuiIngame.java:955)
	at net.minecraftforge.client.GuiIngameForge.renderHotbar(GuiIngameForge.java:206)
	at net.minecraftforge.client.GuiIngameForge.func_73830_a(GuiIngameForge.java:141)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityClientPlayerMP['Axlegear'/913, l='MpServer', x=-3306.44, y=73.62, z=-488.31]]
	Chunk stats: MultiplayerChunkCache: 49, 49
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: World: (132,64,252), Chunk: (at 4,4,12 in 8,15; contains blocks 128,0,240 to 143,255,255), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 120170343 game time, 5932882 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Forced entities: 14 total; [EntityCow['Cow'/973, l='MpServer', x=-3276.50, y=71.00, z=-464.75], EntityBat['Bat'/1070, l='MpServer', x=-3298.25, y=9.10, z=-512.91], EntityBat['Bat'/1071, l='MpServer', x=-3300.75, y=9.10, z=-507.88], EntitySpider['Spider'/720, l='MpServer', x=30.86, y=-45.62, z=-17.67], EntityCow['Cow'/914, l='MpServer', x=-3278.16, y=75.00, z=-494.91], EntityCow['Cow'/915, l='MpServer', x=-3292.78, y=76.00, z=-499.75], EntityClientPlayerMP['Axlegear'/913, l='MpServer', x=-3306.44, y=73.62, z=-488.31], EntityCow['Cow'/916, l='MpServer', x=-3288.38, y=78.00, z=-507.41], EntityBat['Bat'/1012, l='MpServer', x=-3298.44, y=17.62, z=-464.63], EntityCow['Cow'/917, l='MpServer', x=-3295.09, y=74.00, z=-509.06], EntityBat['Bat'/1014, l='MpServer', x=-3318.75, y=17.81, z=-464.53], EntityItem['item.tile.rail'/951, l='MpServer', x=-3303.88, y=20.13, z=-458.69], EntityItem['item.tile.rail'/952, l='MpServer', x=-3295.84, y=21.13, z=-459.13], EntityItem['item.tile.rail'/953, l='MpServer', x=-3301.13, y=21.13, z=-458.13]]
	Retry entities: 0 total; []
	Server brand: fml,forge
	Server type: Non-integrated multiplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:373)
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2444)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:919)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.7.10
	Operating System: Windows 7 (amd64) version 6.1
	Java Version: 1.8.0_25, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 929333088 bytes (886 MB) / 1843011584 bytes (1757 MB) up to 3113877504 bytes (2969 MB)
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmn1G -Xmx3G -XX:+DisableExplicitGC -XX:+UseConcMarkSweepGC -XX:MaxGCPauseMillis=80 -XX:+UseCompressedOops -XX:+UseCodeCacheFlushing -XX:ParallelGCThreads=4
	AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
	IntCache: cache: 0, tcache: 0, allocated: 15, tallocated: 95
	FML: MCP v9.05 FML v7.10.99.99 Minecraft Forge 10.13.4.1492 66 mods loaded, 66 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
	UCHIJA	mcp{9.05} [Minecraft Coder Pack] (minecraft.jar) 
	UCHIJA	FML{7.10.99.99} [Forge Mod Loader] (forge-1.7.10-10.13.4.1492-1.7.10.jar) 
	UCHIJA	Forge{10.13.4.1492} [Minecraft Forge] (forge-1.7.10-10.13.4.1492-1.7.10.jar) 
	UCHIJA	CodeChickenCore{1.0.7.46} [CodeChicken Core] (minecraft.jar) 
	UCHIJA	ivtoolkit{1.2} [IvToolkit] (minecraft.jar) 
	UCHIJA	MobiusCore{1.2.5} [MobiusCore] (minecraft.jar) 
	UCHIJA	NotEnoughItems{1.0.5.111} [Not Enough Items] (NotEnoughItems-1.7.10-1.0.5.111-universal.jar) 
	UCHIJA	psychedelicraftcore{1.4} [Psychedelicraft Core] (minecraft.jar) 
	UCHIJA	AppleCore{1.1.0} [AppleCore] (AppleCore-mc1.7.10-1.1.0.jar) 
	UCHIJA	BiblioCraft{1.10.5} [BiblioCraft] (BiblioCraft[v1.10.6][MC1.7.10].jar) 
	UCHIJA	BuildCraft|Core{7.1.12} [BuildCraft] (buildcraft-7.1.12.jar) 
	UCHIJA	BuildCraft|Builders{7.1.12} [BC Builders] (buildcraft-7.1.12.jar) 
	UCHIJA	BuildCraft|Transport{7.1.12} [BC Transport] (buildcraft-7.1.12.jar) 
	UCHIJA	BuildCraft|Energy{7.1.12} [BC Energy] (buildcraft-7.1.12.jar) 
	UCHIJA	BuildCraft|Silicon{7.1.12} [BC Silicon] (buildcraft-7.1.12.jar) 
	UCHIJA	BuildCraft|Robotics{7.1.12} [BC Robotics] (buildcraft-7.1.12.jar) 
	UCHIJA	BuildCraft|Factory{7.1.12} [BC Factory] (buildcraft-7.1.12.jar) 
	UCHIJA	IC2{2.2.790-experimental} [IndustrialCraft 2] (industrialcraft-2-2.2.790-experimental.jar) 
	UCHIJA	Forestry{3.6.3.20} [Forestry for Minecraft] (forestry_1.7.10-3.6.3.20.jar) 
	UCHIJA	BuildCraft|Compat{7.1.2} [BuildCraft Compat] (buildcraft-compat-7.1.2.jar) 
	UCHIJA	CarpentersBlocks{3.3.7 DEV} [Carpenter's Blocks] (Carpenter's Blocks v3.3.7 DEV - MC 1.7.10.jar) 
	UCHIJA	ForgeMultipart{1.2.0.345} [Forge Multipart] (ForgeMultipart-1.7.10-1.2.0.345-universal.jar) 
	UCHIJA	Railcraft{9.8.0.0} [Railcraft] (Railcraft_1.7.10-9.8.0.0.jar) 
	UCHIJA	chisel{2.4.1.40} [Chisel 2] (Chisel2-2.4.1.40.jar) 
	UCHIJA	ctmlib{1.0.1.5} [CTMLib] (Chisel2-2.4.1.40.jar) 
	UCHIJA	eplus{3.0.2-d} [Enchanting Plus] (EnchantingPlus-1.7.10-3.0.2-d.jar) 
	UCHIJA	ExtraUtilities{1.2.6} [Extra Utilities] (extrautilities-1.2.6.jar) 
	UCHIJA	FastCraft{1.21} [FastCraft] (fastcraft-1.21.jar) 
	UCHIJA	fastleafdecay{1.1} [Fast Leaf Decay] (FastLeafDecay-1.7.10-1.1.jar) 
	UCHIJA	McMultipart{1.2.0.345} [Minecraft Multipart Plugin] (ForgeMultipart-1.7.10-1.2.0.345-universal.jar) 
	UCHIJA	GregsLighting{1.11.2} [Greg's Lighting] (GregsLighting-1.11.2-mc1.7.10.jar) 
	UCHIJA	harvestcraft{1.7.10j} [Pam's HarvestCraft] (Pam's HarvestCraft 1.7.10k.jar) 
	UCHIJA	gregtech{MC1710} [GregTech] (gregtech-5.09.10.jar) 
	UCHIJA	Growthcraft{2.1.0a} [Growthcraft] (growthcraft-core-1.7.10-2.1.0a.jar) 
	UCHIJA	Growthcraft|Cellar{2.1.0a} [Growthcraft Cellar] (growthcraft-cellar-1.7.10-2.1.0a.jar) 
	UCHIJA	Growthcraft|Apples{2.1.0} [Growthcraft Apples] (growthcraft-apples-1.7.10-2.1.0.jar) 
	UCHIJA	Growthcraft|Bamboo{2.1.0} [Growthcraft Bamboo] (growthcraft-bamboo-1.7.10-2.1.0.jar) 
	UCHIJA	Growthcraft|Bees{2.1.0} [Growthcraft Bees] (growthcraft-bees-1.7.10-2.1.0.jar) 
	UCHIJA	Growthcraft|Fishtrap{2.1.0a} [Growthcraft Fishtrap] (growthcraft-fishtrap-1.7.10-2.1.0a.jar) 
	UCHIJA	Growthcraft|Grapes{2.1.0} [Growthcraft Grapes] (growthcraft-grapes-1.7.10-2.1.0.jar) 
	UCHIJA	Growthcraft|Hops{2.1.0} [Growthcraft Hops] (growthcraft-hops-1.7.10-2.1.0.jar) 
	UCHIJA	Growthcraft|Rice{2.1.0a} [Growthcraft Rice] (growthcraft-rice-1.7.10-2.1.0a.jar) 
	UCHIJA	GTTweaker{1.0.9} [GTTweaker] (GTTweaker-1.7.10-1.0.9.jar) 
	UCHIJA	IC2NuclearControl{2.2.5a-GoingForTheKill} [Nuclear Control 2] (IC2NuclearControl-2.2.5a-GoingForTheKill.jar) 
	UCHIJA	ihl{0.579} [IHL Tools & Machines for IC2V2] (ihl-tools_n_machines-0.579-1.7.10.jar) 
	UCHIJA	ImmersiveEngineering{0.5.2} [Immersive Engineering] (ImmersiveEngineering-0.5.2.jar) 
	UCHIJA	ImmibisCore{59.1.2} [Immibis Core] (immibis-core-59.1.2.jar) 
	UCHIJA	Waila{1.5.10} [Waila] (Waila-1.5.10_1.7.10.jar) 
	UCHIJA	JABBA{1.2.1} [JABBA] (Jabba-1.2.1a_1.7.10.jar) 
	UCHIJA	Magneticraft{0.3.3} [Magneticraft] (magneticraft-0.3.3.jar) 
	UCHIJA	MineTweaker3{3.0.10} [MineTweaker 3] (MineTweaker3-1.7.10-3.0.10B.jar) 
	UCHIJA	NEIAddons{1.12.11.36} [NEI Addons] (neiaddons-1.12.11.36-mc1.7.10.jar) 
	UCHIJA	NEIAddons|Developer{1.12.11.36} [NEI Addons: Developer Tools] (neiaddons-1.12.11.36-mc1.7.10.jar) 
	UCHIJA	NEIAddons|AppEng{1.12.11.36} [NEI Addons: Applied Energistics 2] (neiaddons-1.12.11.36-mc1.7.10.jar) 
	UCHIJA	NEIAddons|Botany{1.12.11.36} [NEI Addons: Botany] (neiaddons-1.12.11.36-mc1.7.10.jar) 
	UCHIJA	NEIAddons|Forestry{1.12.11.36} [NEI Addons: Forestry] (neiaddons-1.12.11.36-mc1.7.10.jar) 
	UCHIJA	NEIAddons|CraftingTables{1.12.11.36} [NEI Addons: Crafting Tables] (neiaddons-1.12.11.36-mc1.7.10.jar) 
	UCHIJA	NEIAddons|ExNihilo{1.12.11.36} [NEI Addons: Ex Nihilo] (neiaddons-1.12.11.36-mc1.7.10.jar) 
	UCHIJA	neiintegration{1.0.12} [NEI Integration] (NEIIntegration-MC1.7.10-1.0.12.jar) 
	UCHIJA	MapWriter{2.1.2} [MapWriter] (Opis-1.2.5_1.7.10.jar) 
	UCHIJA	Opis{1.2.5} [Opis] (Opis-1.2.5_1.7.10.jar) 
	UCHIJA	psychedelicraft{1.5.2} [Psychedelicraft] (Psychedelicraft-1.5.2.n118.jar) 
	UCHIJA	RedLogic{59.1.11} [RedLogic] (redlogic-59.1.11.jar) 
	UCHIJA	SpecialMobs{3.1.1} [Special Mobs] (SpecialMobs-1.7.10-3.1.1.jar) 
	UCHIJA	StacksOnStacks{{$version}} [Stacks on Stacks!] (StacksOnStacks-1.7.10-2.2.0.jar) 
	UCHIJA	ForgeMicroblock{1.2.0.345} [Forge Microblocks] (ForgeMultipart-1.7.10-1.2.0.345-universal.jar) 
	GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.5.0 NVIDIA 355.98' Renderer: 'GeForce GTX 660/PCIe/SSE2'
	Forestry : Info: The following plugins have been disabled in the config: lepidopterology
	Launched Version: 1.7.10-Forge10.13.4.1492-1.7.10
	LWJGL: 2.9.1
	OpenGL: GeForce GTX 660/PCIe/SSE2 GL version 4.5.0 NVIDIA 355.98, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Anisotropic filtering is supported and maximum anisotropy is 16.
Shaders are available because OpenGL 2.1 is supported.

	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: []
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	Vec3 Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
	Anisotropic Filtering: Off (1)

**Comment by axlegear** [comment]
Looks like IC2, tho.

**Comment by axlegear** [comment]
http://bt.industrial-craft.net/view.php?id=1763

**Comment by Blood-Asp** [comment]
You did regenerate the world in beween the restarts with different mods? Because the blocks that are in there trough changed IDs do not indicate the mod that broke the worldgen.

**Comment by axlegear** [comment]
Regenerated chunks, yes.  When testing with only IC2+GT, I went into SP and generated, wandered around a while, and found continual fuckage.  You can see pictures on the post on MantisBT for the IC2 thread.

**Comment by Blood-Asp** [comment]
The crash report and the items in hotbar show clearly that there is NOT only GT+IC2 in the world. So what are you talking about?

**Comment by axlegear** [comment]
You're misunderstanding and reading too far into the pics.  I tested with GT and IC2 only and it still occured.  The pics are in the native state and there to show the corruption of the chunk.

I'm not sure how I can be more clear.  I removed mods one by one, as per my history above in this very thread.  The problem persists through every condition.  The only case that it works if if I removed either GT5U, or both GT5U and IC2.  Having IC2 without GT5U seems okay.
